spec/unit/plugins/passwd_spec.rb

```diff
@@ -16,50 +16,235 @@
+        .with("get-localgroup | convertto-json")
+        .and_return(mock_shell_out(0, group_info.to_json, ""))
+    end
+
+    def transform(user_data)
+      Hash[
+        user_data.map do |key, val|
+          [key.downcase, val]
+        end
+      ]
+    end
+
+    it "returns lower-cased passwd keys for each local user" do
+      {
+        "groupone" => group_one_info.to_json,
+        "grouptwo" => group_two_info.to_json,
+      }.each do |gname, info|
+        expect(plugin).to receive(:powershell_out)
+          .with("get-localgroupmember -name '#{gname}' | convertto-json")
+          .and_return(mock_shell_out(0, "[]", ""))
+      end
+      plugin.run
+      expect(plugin[:etc][:passwd].keys.sort).to eq(%w{userone usertwo}.sort)
+    end
+
+    it "returns preserved-case passwd entries for local users" do
+      {
+        "groupone" => group_one_info.to_json,
+        "grouptwo" => group_two_info.to_json,
+      }.each do |gname, info|
+        expect(plugin).to receive(:powershell_out)
+          .with("get-localgroupmember -name '#{gname}' | convertto-json")
+          .and_return(mock_shell_out(0, "[]", ""))
+      end
+      plugin.run
+      expect(plugin[:etc][:passwd]["userone"]).to eq(transform(user_info[0]))
+    end
+
+    it "returns lower-cased group entries for each local group" do
+      {
+        "groupone" => group_one_info.to_json,
+        "grouptwo" => group_two_info.to_json,
+      }.each do |gname, info|
+        expect(plugin).to receive(:powershell_out)
+          .with("get-localgroupmember -name '#{gname}' | convertto-json")
+          .and_return(mock_shell_out(0, "[]", ""))
+      end
+      plugin.run
+      expect(plugin[:etc][:group].keys.sort).to eq(%w{groupone grouptwo}.sort)
+    end
+
+    it "returns preserved-cased group entries for local groups" do
+      {
+        "groupone" => group_one_info.to_json,
+        "grouptwo" => group_two_info.to_json,
+      }.each do |gname, info|
+        expect(plugin).to receive(:powershell_out)
+          .with("get-localgroupmember -name '#{gname}' | convertto-json")
+          .and_return(mock_shell_out(0, "[]", ""))
+      end
+      plugin.run
+      expect(plugin[:etc][:group]["grouptwo"]).to eq(
+        transform(group_info[1]).merge({ "members" => [] })
+      )
+    end
+
+    it "returns members for groups" do
+      {
+        "groupone" => group_one_info.to_json,
+        "grouptwo" => group_two_info.to_json,
+      }.each do |gname, info|
+        expect(plugin).to receive(:powershell_out)
+          .with("get-localgroupmember -name '#{gname}' | convertto-json")
+          .and_return(mock_shell_out(0, info, ""))
+      end
       plugin.run
-      root = plugin[:etc][:passwd]["root"]
-      expect(root["gecos"].encoding).to eq(Encoding.default_external)
+      expect(plugin[:etc][:group]["groupone"]["members"]).to eq(group_one_info)
+      expect(plugin[:etc][:group]["grouptwo"]["members"]).to eq(group_two_info)
     end
   end
 end
```

Comment:
Given our confusion around the unescaping it'd probably be a good idea to add a test that that's happening properly.

Reply:
Definitely. As I said, I hadn't updated any of the tests for this, so will do that later today.

Reply:
Oh oops, I didn't see that part.